### PR TITLE
fix(ui) : resolved warnings after selecting a service groups in an inexistant timeperiod

### PR DESCRIPTION
**Pull request by AssanDiomande**
## Description

Fixes MON-11771

* Fixes variable names mismatch
* The service start date and end date between the website form and the database (table log_archive_service from centreon_storage) doesn't match which leads to the fact that the sql request return an empty response , so I added a condition in this case (set undertemined to 100 and gather host and service information)
* Enable redirection to a specific host from resource status page

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.10.x
- [ ] 21.04.x
- [x] 21.10.x
- [x] 22.04.x (master)

<h2> How this pull request can be tested ? </h2>

* create a empty service groups 
* create a service groups with a least one service
* check if there is warnings/errors when selecting an host groups / service groups

### CODE REVIEW

File: www/include/reporting/dashboard/DB-Func.php [hunk 1]
```diff
@@ -595,10 +595,20 @@ function getLogInDbForServicesGroup($servicegroupId, $startDate, $endDate, $repo
             foreach ($serviceStatsLabels as $name) {
                 $serviceGroupStats["average"][$name] += $servicesStats[$hostId][$serviceId][$name];
             }
+        } else {
+            $serviceGroupStats["average"]["UNDETERMINED_TP"] = 100;
+            $serviceGroupStats[$hostServiceid]["HOST_ID"] = $hostId;
+            $serviceGroupStats[$hostServiceid]["SERVICE_ID"] = $serviceId;
+            $serviceGroupStats[$hostServiceid]["HOST_NAME"] = $service['host_name'];
+            $serviceGroupStats[$hostServiceid]["SERVICE_DESC"] = $service['service_description'];
```
Comment (kduret) [inline www/include/reporting/dashboard/DB-Func.php:603]: please, do not duplicate code
you are doing exactly the same thing than L591-L594

**Review by kduret — CHANGES_REQUESTED** [review]

File: www/include/reporting/dashboard/DB-Func.php (hunk 1 above)
Comment (AssanDiomande) [reply www/include/reporting/dashboard/DB-Func.php]: done

**Review by kduret — DISMISSED** [review]

File: www/include/reporting/dashboard/template/viewHostGroupLog.ihtml [hunk 2]
```diff
@@ -33,7 +33,7 @@
 	<tr class='list_lvl_3'>
 		<td style='width:300px; padding:5px'>
 			<!-- chart drawing -->
-			{if $id}
+			{if $hostgroup_id}
```
Comment (jeremyjaouen) [inline www/include/reporting/dashboard/template/viewHostGroupLog.ihtml:36]:
```suggestion
			{if isset($hostgroup_id)}
```

Comment (jeremyjaouen) [reply www/include/reporting/dashboard/template/viewHostGroupLog.ihtml:36]: `{if $hostgroup_id}` could throw a warning in logs if the variable doesn't exist

**Review by jeremyjaouen — CHANGES_REQUESTED** [review]

File: www/include/reporting/dashboard/template/viewServicesGroupLog.ihtml [hunk 3]
```diff
@@ -35,7 +35,7 @@
 	<tr class='list_lvl_3'>
 		<td style='width:300px; padding:5px'>
 			<!-- chart drawing -->
-			{if $id}
+			{if $servicegroup_id}
```
Comment (jeremyjaouen) [inline www/include/reporting/dashboard/template/viewServicesGroupLog.ihtml:38]:
```suggestion
			{if isset($servicegroup_id)}
```

**Review by jeremyjaouen — CHANGES_REQUESTED** [review]

File: www/include/reporting/dashboard/template/viewServicesGroupLog.ihtml (hunk 3 above)
Comment (AssanDiomande) [reply www/include/reporting/dashboard/template/viewServicesGroupLog.ihtml]: updated

File: www/include/reporting/dashboard/template/viewHostGroupLog.ihtml (hunk 2 above)
Comment (AssanDiomande) [reply www/include/reporting/dashboard/template/viewHostGroupLog.ihtml]: updated

**Review by kduret — APPROVED** [review]

**Review by jeremyjaouen — APPROVED** [review]

**Pull request merged**